File: user-service/src/main/java/rs/raf/user_service/domain/entity/VerificationRequest.java
package rs.raf.user_service.domain.entity;


import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import rs.raf.user_service.domain.enums.VerificationStatus;
import rs.raf.user_service.domain.enums.VerificationType;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class VerificationRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;
    private Long targetId;

    @Enumerated(EnumType.STRING)
    private VerificationStatus status; // PENDING, APPROVED, DENIED

    @Enumerated(EnumType.STRING)
    private VerificationType verificationType; // LOGIN, LOAN

    private LocalDateTime expirationTime;

    @CreationTimestamp
    private LocalDateTime createdAt;

    private String details;
}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/domain/entities/VerificationRequest.kt
package pack.userservicekotlin.domain.entities

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import pack.userservicekotlin.domain.enums.VerificationStatus
import pack.userservicekotlin.domain.enums.VerificationType
import java.time.LocalDateTime

@Entity
open class VerificationRequest(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null,
    open var userId: Long? = null,
    open var targetId: Long? = null,
    @Enumerated(EnumType.STRING)
    open var status: VerificationStatus? = null,
    @Enumerated(EnumType.STRING)
    open var verificationType: VerificationType? = null,
    open var expirationTime: LocalDateTime? = null,
    @CreationTimestamp
    open var createdAt: LocalDateTime? = null,
    open var details: String? = null,
)
